src/assignments/assignments.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from "@nestjs/common";
+
+import { ClassroomsRepository } from "@/classrooms/classrooms.repository";
+import { FileUploadsService } from "@/file-uploads/file-uploads.service";
+
+import { CreateAssignmentDto } from "./assignments.dtos";
+import { AssignmentsRepository } from "./assignments.repository";
+
+@Injectable()
+export class AssignmentsService {
+  constructor(
+    private readonly assignmentsRepository: AssignmentsRepository,
+    private readonly fileUploadsService: FileUploadsService,
+    private readonly classroomsRepository: ClassroomsRepository,
+  ) {}
+
+  async uploadAssignment(
+    file: Express.Multer.File,
+    classroomId: number,
+    title: string,
+    description: string,
+    dueDate: Date,
+  ) {
+    const classroom = await this.classroomsRepository.findOneOrFail({ id: classroomId });
+
+    const presignedUrl = await this.fileUploadsService.getPresignedUrl(file);
+
+    const uploadResponse = await this.fileUploadsService.uploadToS3(presignedUrl, file);
+
+    if (!uploadResponse.ok) {
+      throw new Error("Failed to upload file to S3");
+    }
+
+    const createAssignmentDto: CreateAssignmentDto = {
+      title,
+      description,
+      fileUrl: presignedUrl.split("?")[0],
```

Comment:
Shouldn't the uploadResponse send back a url for the file?

Reply:
Yes, it was a typo from my end, but somehow the presignedUrl was working.